src/Simulation/DepletionVoltage.jl

```diff
@@ -0,0 +1,163 @@
+            start_local_search = T(U)
+            break
+        end
+    end
+    local_range::AbstractRange = range(start_local_search, last(potential_range), step = step(potential_range))
+    size_data = size(sim.point_types.data)
+    
+    if size_data[2] == 1 # Fully phi symmetric detectors
+        Ix = CartesianIndex(1,0,0)
+        Iz = CartesianIndex(0,0,1)
+        neighbours = [Ix,-Ix,Iz,-Iz]
+    else
+        Ix = CartesianIndex(1,0,0)
+        Iy = CartesianIndex(0,1,0)
+        Iz = CartesianIndex(0,0,1)
+        neighbours = [Ix,-Ix,Iy,-Iy,Iz,-Iz]
+    end
+
+    scale = start_local_search
+    undepleted = 0
+    eps = 1e-1
+    length_idx = length(inside[1])
+    for (i,idx) in enumerate(inside)
+        for scale_loc in local_range
+            center_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[idx] + ϕρ[idx]
+            min_pot = center_pot + eps
+            max_pot = center_pot - eps
+            for neighbour in neighbours
+                n_idx = idx + neighbour
+                if 0 == sum([ (n_idx[j] == 0 || n_idx[j] > size_data[j]) for j in 1:length_idx]) # Check whether one of the indices is outside of the grid
+                    local_pot = T(scale_loc) * sim.weighting_potentials[contact_id].data[n_idx] + ϕρ[n_idx]
+                    if local_pot < min_pot
+                        min_pot = local_pot
+                    elseif local_pot > max_pot
+                        max_pot = local_pot
+                    end
+                end
+            end
+            if !(min_pot>center_pot || max_pot<center_pot) # Check whether depletion condition is met for this specific index (corresponding to center pot) and jump to next index if that is the case.
+                if abs(scale_loc)> abs(scale)
+                    scale = T(scale_loc)
+                    local_range = range(scale, last(potential_range), step = step(potential_range))
+                end         
+                break
+            end
+            if scale_loc == local_range[end] # If local scale is iterated all the way throughout the local range, this grid point is undepleted.
+                undepleted+=1
+            end         
+        end
+        if (undepleted > 0) # Break loop as soon as one grid point is undepleted
+            break
+        end
+    end
+    if initial_depletion
+        @info "Detector is already depleted at the start of the specified voltage range ($(potential_range))! Please find a starting voltage where the detector is not yet depleted."
+    elseif undepleted == 0
+        depletion_voltage = scale
+    end
+    
+    if verbose
+        if !isnan(depletion_voltage)
+            @info "The depletion voltage of the detector is ($(depletion_voltage) ± $(T(step(potential_range)))) V applied to contact $(contact_id)."
+        else
+            @warn "The depletion voltage is not in the specified range $(extrema(potential_range).*u"V")."
```

Comment:
This is something that should already be checked at the beginning of the function where the depletion status at the edges of the `potential_range` are checked. If they are equal: print this warning and already return `NaN`.